lib/livebook_web/live/session_live/secrets_component.ex

```diff
@@ -1,6 +1,25 @@
 defmodule LivebookWeb.SessionLive.SecretsComponent do
   use LivebookWeb, :live_component
 
+  @impl true
+  def mount(socket) do
+    {:ok, assign(socket, initialized: false)}
+  end
+
+  @impl true
+  def update(assigns, socket) do
+    socket = assign(socket, assigns)
+
+    socket =
+      if socket.assigns.initialized do
+        socket
+      else
+        assign(socket, data: %{"label" => assigns.secret, "value" => ""}, initialized: true)
+      end
+
+    {:ok, socket}
+  end
```

Comment:
Ah, we can check for the `:data` assign itself, like this:

```suggestion
  @impl true
  def update(assigns, socket) do
    socket = assign(socket, assigns)

    socket =
      if socket.assigns[:data] do
        socket
      else
        assign(socket, data: %{"label" => assigns.secret, "value" => ""})
      end

    {:ok, socket}
  end
```